pkg/component/etcd/etcd.go

```diff
@@ -584,6 +592,81 @@ func (e *etcd) emptyHVPA() *hvpav1alpha1.Hvpa {
 	return &hvpav1alpha1.Hvpa{ObjectMeta: metav1.ObjectMeta{Name: e.etcd.Name, Namespace: e.namespace}}
 }
 
+func (e *etcd) emptyVerticalPodAutoscaler() *vpaautoscalingv1.VerticalPodAutoscaler {
+	return &vpaautoscalingv1.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: e.etcd.Name + "-vpa", Namespace: e.namespace}}
+}
+
+func (e *etcd) reconcileVerticalPodAutoscaler(ctx context.Context, vpa *vpaautoscalingv1.VerticalPodAutoscaler, minAllowed, maxAllowed corev1.ResourceList) error {
+	vpaUpdateMode := vpaautoscalingv1.UpdateModeAuto
+	containerPolicyOff := vpaautoscalingv1.ContainerScalingModeOff
+	containerPolicyAuto := vpaautoscalingv1.ContainerScalingModeAuto
+	controlledValues := vpaautoscalingv1.ContainerControlledValuesRequestsOnly
+
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, vpa, func() error {
+		var scaleDownUpdateMode *string
+
+		vpaLabels := vpa.GetLabels()
+		if vpaLabels == nil {
+			vpaLabels = map[string]string{}
+		}
+		vpaAnnotations := vpa.GetAnnotations()
+		if vpaAnnotations == nil {
+			vpaAnnotations = map[string]string{}
+		}
```

Comment:
Nit: You could drop this and use `metav1.SetMetadata{Label,Annotation}` instead.

Reply:
done.